CNV/gCNV/codex/resources/codex_vcf_converter.py

```diff
@@ -0,0 +1,191 @@
+variant_standard_field_keys = ['CHROM', 'POS', 'ID', 'REF', 'ALT',
+                               'QUAL', 'FILTER', 'INFO', 'FORMAT']
+
+
+def _convert_call_fields_to_variant(
+        variant_id: str,
+        call_fields_per_variant_sample: List[Dict[str, str]],
+        sample_names: List[str]) -> Tuple[Dict[str, str], List[str]]:
+    # parse call fields from first sample for standard variant fields
+    contig = call_fields_per_variant_sample[0]['chr']
+    start = call_fields_per_variant_sample[0]['st_bp']
+
+    # construct variant standard fields
+    num_samples = len(sample_names)
+    info_field = _construct_info_field(call_fields_per_variant_sample,
+                                       num_samples)
+    variant_standard_fields = OrderedDict(zip(
+        variant_standard_field_keys,
+        [contig, start, variant_id, 'N', '<DEL>,<DUP>', '.', '.', info_field,
+         'GT:GQ:PL']))
+
+    # construct list of per-sample genotype fields
+    # (we do not use a dict here to allow for duplicate sample names)
+    # for all non-variant samples, the genotype field is '.'
+    # for all variant samples, we construct the corresponding genotype field
+    sample_to_genotype_field_map = defaultdict(
+        lambda: '.',
+        zip([call_fields['sample_name']
+             for call_fields in call_fields_per_variant_sample],
+            map(_construct_genotype_field, call_fields_per_variant_sample)))
+    genotype_fields = [sample_to_genotype_field_map[sample_name]
+                       for sample_name in sample_names]
+
+    return variant_standard_fields, genotype_fields
+
+
+def _construct_info_field(
+        call_fields_per_variant_sample: List[Dict[str, str]],
+        num_samples: int) -> str:
+    # calculate ACs and AFs
+    cnv_fields_per_variant_sample = \
+        [call_fields['cnv']
+         for call_fields in call_fields_per_variant_sample]
+    cnv_counter = Counter(cnv_fields_per_variant_sample)
+    ac_del = cnv_counter['del']
+    ac_dup = cnv_counter['dup']
+    an = len(cnv_fields_per_variant_sample)
+    af_del = '{0:.3f}'.format(float(ac_del) / num_samples)
+    af_dup = '{0:.3f}'.format(float(ac_dup) / num_samples)
+
+    # parse call fields from first sample
+    call_fields = call_fields_per_variant_sample[0]
+    end = call_fields['ed_bp']
+    start_exon = int(call_fields['st_exon'])
+    end_exon = int(call_fields['ed_exon'])
+    num_targets = end_exon - start_exon + 1
+
+    # construct info field
+    info_field = \
+        'AC=' + str(ac_del) + ',' + str(ac_dup) + ';' \
+              + 'AF=' + str(af_del) + ',' + str(af_dup) + ';' \
+              + 'AN=' + str(an) + ';' \
+              + 'END=' + str(end) + ';' \
+              + 'NT=' + str(num_targets)
+
+    return info_field
+
+
+def _construct_genotype_field(
+        call_fields: Dict[str, str]) -> str:
+    # parse call fields
+    cnv = call_fields['cnv']
+    lratio = int(float(call_fields['lratio']))
```

Comment:
We had some troubles interpreting the results of the seg file in the gCNV Spark evaluations. Here what their documentation says: lratio (likelihood ratio of CNV event versus copy neutral event). So it should be a proxy for GQ, however it yields strange results..